api/validation.go

```diff
@@ -1356,3 +1384,44 @@ func validateBucketName(s string) error {
 
 	return nil
 }
+
+// Returns `true` if no backups are running for the specified cluster.
+func (e *EverestServer) ensureNoBackupsRunningForCluster(ctx context.Context, dbClusterName, namespace string) (bool, error) {
+	backupList, err := e.kubeClient.ListDatabaseClusterBackups(ctx, namespace, metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(&metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				databaseClusterNameLabel: dbClusterName,
+			},
+		}),
+	})
+	if err != nil {
+		return false, errors.Join(err, errors.New("could not list Database Cluster Backups"))
+	}
+	return !slices.ContainsFunc(backupList.Items, func(b everestv1alpha1.DatabaseClusterBackup) bool {
+		return (b.Status.State == everestv1alpha1.BackupRunning ||
+			b.Status.State == everestv1alpha1.BackupStarting ||
+			b.Status.State == everestv1alpha1.BackupNew) &&
+			b.DeletionTimestamp.IsZero()
+	}), nil
+}
+
+// isBackupScheduleRunning returns true if a backup schedule is running at the current time.
+func isBackupScheduleRunning(
+	currTime time.Time,
+	backupSchedules []everestv1alpha1.BackupSchedule,
+) (bool, error) {
+	currTime = currTime.Truncate(time.Minute)
```

Comment:
The `.Truncate()` function removes the seconds like `16:05:01 -> 16:05:00, 16:05:59 -> 16:05:00`, is that the intended behaviour? Shouldn't the `.Add(-1 * time.Minute)` be used here?

Reply:
`Truncate()` is intended here. In this case, it truncates to the nearest minute, so it removes everything from the seconds. Reason for this is that `NextTickAfter` does not return seconds, so the `Equals` check will fail.

> Shouldn't the .Add(-1 * time.Minute) be used here?

That won't work here. If we do this, we are simply avoiding creation of backups 1minute before the actual schedule. But what happens when the schedule is triggered? This check will allow us to create an on-demand backup even though a `psmdb-backup` object has been created (note that dbbackup object creation will be delayed for a few seconds). This won't fix the issue..

Reply:
Ok, I see. Thanks, the concern is resolved 👍